<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UiCertificates</class>
 <widget class="QWidget" name="UiCertificates">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>745</width>
    <height>518</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>X509 Certs &amp; CSR Tool</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QTabWidget" name="certificates_tabWidget_X25519">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabName_certificates_X25519_Parse">
      <attribute name="title">
       <string>Parse X509 Cert</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_certificates_Parse_LoadCertificate">
         <property name="text">
          <string>Load certificate</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>349</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0" colspan="6">
        <widget class="QSplitter" name="splitter_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <widget class="QWidget" name="verticalLayoutWidget_4">
          <layout class="QVBoxLayout" name="verticalLayout_Input_2">
           <item>
            <widget class="QLabel" name="label_27">
             <property name="text">
              <string>Input certificate (PEM/HEX/File)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="AdvancedTextEdit" name="textEdit_certificates_Parse_InputCertificate">
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'MS Shell Dlg 2'; font-size:8.25pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="verticalLayoutWidget_3">
          <layout class="QVBoxLayout" name="verticalLayout_Output_2">
           <item>
            <widget class="QLabel" name="label_28">
             <property name="text">
              <string>Parsed certificate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextEdit" name="textEdit_certificates_Parse_ParsedCertificate"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="pushButton_certificates_Parse_ExportCRT">
         <property name="text">
          <string>Save As</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_certificates_Parse_ParseCertificate">
         <property name="text">
          <string>Parse certificate</string>
         </property>
        </widget>
       </item>
       <item row="1" column="5">
        <widget class="QPushButton" name="pushButton_certificates_Parse_CopyCertHEX">
         <property name="text">
          <string>Copy HEX</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QCheckBox" name="checkBox_CertOpensslOutput">
         <property name="text">
          <string>OpenSSL Output (require openssl to be installed)</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabName_Certificates_X25519_ParseCSR">
      <attribute name="title">
       <string>Parse X509 CSR</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" colspan="6">
        <widget class="QSplitter" name="splitter">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <widget class="QWidget" name="verticalLayoutWidget_2">
          <layout class="QVBoxLayout" name="verticalLayout_Input">
           <item>
            <widget class="QLabel" name="label_29">
             <property name="text">
              <string>Input CSR (PEM/HEX/File)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="AdvancedTextEdit" name="textEdit_certificates_ParseCsr_InputCsr">
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'MS Shell Dlg 2'; font-size:8.25pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="verticalLayoutWidget">
          <layout class="QVBoxLayout" name="verticalLayout_Output">
           <item>
            <widget class="QLabel" name="label_30">
             <property name="text">
              <string>Parsed CSR</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextEdit" name="textEdit_certificates_Parse_ParsedCSR"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_certificates_Parse_ParseCSR">
         <property name="text">
          <string>Parse CSR</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_certificates_Parse_LoadCSR">
         <property name="text">
          <string>Load CSR</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="pushButton_certificates_Parse_ExportCSR">
         <property name="text">
          <string>Save As</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>379</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="5">
        <widget class="QPushButton" name="pushButton_certificates_Parse_CopyCSRHEX">
         <property name="text">
          <string>Copy HEX</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QCheckBox" name="checkBox_CsrOpenSslOutput">
         <property name="text">
          <string>OpenSSL Output (require OpenSSL installed)</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>AdvancedTextEdit</class>
   <extends>QTextEdit</extends>
   <header>base/extensions/AdvancedTextEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
